src/main/java/org/apache/datasketches/req/ReqSketch.java

```diff
@@ -336,23 +323,17 @@ public double getRankLowerBound(final double rank, final int numStdDev) {
 
   @Override
   public double[] getRanks(final float[] values) {
-    return getRanks(values, ltEq);
+    return getRanks(values, ltEq == true ? INCLUSIVE : NON_INCLUSIVE);
   }
 
   @Override
-  public double[] getRanks(final float[] values, final boolean inclusive) {
+  public double[] getRanks(final float[] values, final QuantileSearchCriteria searchCrit) {
     if (isEmpty()) { return null; }
+    refreshSortedView();
```

Comment:
Why is the sorted view refreshed on read-only functions?  Would it be better to offset the cost on updates to the sketch or is this necessary because the sketch may be backed by direct memory and modified behind the scenes?

Reply:
Building the sorted view can be an expensive operation.  Once it has been built, and if the sketch has not been updated or been the target of a merge, all subsequent calls to the query methods (of which, getRanks is one) are quite fast.  However, if the user does a getRanks(); update(); getRanks(), the update() call invalidates the internal state of the sketch, so the sorted view needs to be rebuilt.

Thus each of the query calls must check if the internal state of the sketch has changed.  If not, this refreshSortedView() does nothing.